enhancements/ingress/global-options-enable-hsts.md

```diff
@@ -0,0 +1,466 @@
+```go
+// These are strings instead of enums so that they can be passed in easily from the HAProxy template
+const (
+    AllHSTSHeaderPolicyScope         = "All"
+    LimitedHSTSHeaderPolicyScope     = "Limited"
+)
+
+// HSTSDirectiveType is an optional directive that directs HSTS Policy configuration.
+//+kubebuilder:validation:Enum=preload;includeSubDomains
+type HSTSDirectiveType string
+const(
+    // PreloadHSTSDirective directs the client to include hosts in its host preload list so that
+    // it never needs to do an initial load to get the HSTS header (note that this is not defined
+    // in RFC 6797 and is therefore client implementation-dependent).
+    PreloadHSTSDirective             HSTSDirectiveType = "preload"
+
+    // IncludeSubDomainsHSTSDirective means the HSTS Policy applies to any subdomains of the host's
+    // domain name, when the route admission policy allows wildcards.  If wildcard routes aren't allowed,
+    // this directive has no effect.
+    IncludeSubDomainsHSTSDirective   HSTSDirectiveType = "includeSubDomains"
+)
+
+type IngressControllerHSTSHeaderPolicy {
+	// maxAgeSeconds is the delta time range in seconds during which hosts are regarded as HSTS hosts.
+	// If set to 0, it negates the effect and hosts are no longer regarded as HSTS hosts.
+	// maxAgeSeconds is a time-to-live value, and if this policy is not refreshed on a client, the HSTS
+	// policy will eventually expire on that client.
+	// +required
+    MaxAgeSeconds       int32 `json:"maxAgeSeconds"`
+
+    // HSTSDirectives is a list of optional HSTS directives that direct HSTS Policy configuration.
+    // +kubebuilder:validation:UniqueItems=true
+    // +optional
+    HSTSDirectives      []HSTSDirectiveType `json:"hstsDirectives,omitempty"`
+
+    // scope indicates whether this header policy is enabled for a limited set of routes that have hosts
+    // with domains in the list specified in inScopeDomains (limited scope) or for all routes that have
+    // hosts under the cluster ingress domain (all scope).  If HSTS is desired for all routes, set the scope
+    // field to "All". If HSTS is desired for only a limited set of routes, set the scope field to
+    // "Limited" and add a list of subdomains to the InScopeDomains field.  If scope is omitted, no HSTS
+    // header policy will be added.
+    // +optional
+    Scope               string `json:"scope,omitempty"`
+
+    // inScopeDomains is a list of router domains that require HSTS, when not all routes require HSTS.
+    // If scope = All then this field is ignored.  If scope = Limited, HSTS policy is limited to hosts with
+    // domains in the inScopeDomains list.
+    // +optional
+    InScopeDomains      []string `json:"inScopeDomains,omitempty"`
+}
+```
+
+Validation and formatting in the `IngressController`
+would change the fields into a semi-colon delimited string with formatting:
+```yaml
+max-age=<numberofseconds>[;includeSubDomains[;preload]]`
```

Comment:
`preload` could be specified without `includeSubDomains`, right?
```suggestion
max-age=<numberofseconds>[;includeSubDomains][;preload]`
```
(That suggestion still isn't exactly right because the ordering doesn't really matter, but it's close enough.)